### PR TITLE
fix(datahub-upgrade): Add support for Postgres migration

### PR DESCRIPTION
**Pull request by jjoyce0510**
This PR introduces datahub-upgrade SQL suitable for Postgres. Previously, we had only supported MySQL and Maria officially. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java [hunk 1]
```diff
@@ -7,9 +7,16 @@
 import io.ebean.EbeanServer;
 import java.util.function.Function;
 
-// Do we need SQL-tech specific migration paths?
```
Comment (frsann) [inline datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java:10]: Heh, I think we got our answer 😅

File: docker/datahub-upgrade/README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ As of today, there are 2 supported upgrades:
 to metadata_aspect_v2 table. Arguments:
     - *batchSize* (Optional): The number of rows to migrate at a time. Defaults to 1000.
     - *batchDelayMs* (Optional): The number of milliseconds of delay between migrated batches. Used for rate limiting. Defaults to 250. 
+    - *dbType* (optional): The target DB type. Valid values are `MY_SQL`, `MARIA`, `POSTGRES`.
```
Comment (frsann) [inline docker/datahub-upgrade/README.md:13]: If its optional, which is the default?

**Review by frsann — APPROVED** [review]
:lgtm:

**Review by shirshanka — APPROVED** [review]
LGTM

File: docker/datahub-upgrade/README.md (hunk 2 above)
Comment (shirshanka) [reply docker/datahub-upgrade/README.md:13]: +1

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java (hunk 1 above)
Comment (jjoyce0510) [reply datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java:10]: lol. yes!

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java [hunk 3]
```diff
@@ -29,18 +36,45 @@ public int retryCount() {
...
+          : DbType.MYSQL;
+
+      String sqlUpdateStr;
+
+      switch (targetDbType) {
+        case POSTGRES:
+          sqlUpdateStr = "CREATE TABLE IF NOT EXISTS metadata_aspect_v2 (\n"
+              + "  urn                           varchar(500) not null,\n"
+              + "  aspect                        varchar(200) not null,\n"
+              + "  version                       bigint(20) not null,\n"
```
Comment (frsann) [inline datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java:51]: After some testing, it seems that `bigint(20)`  is not supported in Postgres. The equivalent is simply `bigint`, it seems.

**Review by frsann — CHANGES_REQUESTED** [review]

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java (hunk 3 above)
Comment (jjoyce0510) [reply datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java]: Yes, you're right. Just also verified. Thanks!

**Review by frsann — APPROVED** [review]

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java [hunk 4]
```diff
@@ -29,18 +36,45 @@ public int retryCount() {
...
+          sqlUpdateStr = "CREATE TABLE IF NOT EXISTS metadata_aspect_v2 (\n"
+              + "  urn                           varchar(500) not null,\n"
+              + "  aspect                        varchar(200) not null,\n"
+              + "  version                       bigint not null,\n"
+              + "  metadata                      text not null,\n"
+              + "  systemmetadata                text,\n"
+              + "  createdon                     timestamp not null,\n"
+              + "  createdby                     varchar(255) not null,\n"
+              + "  createdfor                    varchar(255),\n"
+              + "  constraint pk_metadata_aspect primary key (urn,aspect,version)\n"
```
Comment (frsann) [inline datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/CreateAspectTableStep.java:57]:
```suggestion
              + " primary key (urn,aspect,version)\n"
```

**Review by frsann — CHANGES_REQUESTED** [review]

**Review by frsann — APPROVED** [review]

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**